### ISSUE TITLE
Nan Loss after a couple of epochs

### ISSUE THREAD

**Issue by Yarp96** [shariqfarooq123/AdaBins#16]
Hi,

First of all very nice paper and architechture! Ive been playing with the training code and trained a model with my own data. However after a couple of epochs im getting a Nan loss and after some debugning it seems like its the output of the model returns a nan tensor. I played a little bit with gradient clipping to see if it would help but it does not seem to be the case.

Best regards

**Comment by hahawhat-ch** [comment]
> Hi,
> 
> First of all very nice paper and architechture! Ive been playing with the training code and trained a model with my own data. However after a couple of epochs im getting a Nan loss and after some debugning it seems like its the output of the model returns a nan tensor. I played a little bit with gradient clipping to see if it would help but it does not seem to be the case.
> 
> Best regards

Hi
Have you found a solution.I'm facing the same problem

**Comment by hahawhat-ch** [comment]
I think it's the problem. I add the min depth as you recommend. then run a epoch to see the result.the loss have value to decrease. thank u！



---Original---
From: "Shariq F. ***@***.***&gt;
Date: Tue, Mar 9, 2021 21:39 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [shariqfarooq123/AdaBins] Nan Loss after a couple of epochs (#16)




 
One reason might be that at some point loss has a 'nan' value. What is the minimum depth you are using?
 The code uses SILog and if your depth has zeroes (or negative) this will lead to nan values.
 
If that's the case, you can:
  
Clip (from below) your depth values to some minimum positive value (like 1e-3). 

You can change the loss in line 19 in loss.py (https://github.com/shariqfarooq123/AdaBins/blob/2fb686a66a304f0a719bc53d77412460af97fd61/loss.py#L19)
 to : 
 alpha = 1e-3 g = torch.log(alpha + input) - torch.log(alpha + target)
 
Let me know if this solves the issues.
